src/modules/login.js

```diff
@@ -0,0 +1,287 @@
+    marginTop: '40px',
+    fontFamily: "'PT Sans', sans-serif",
+    fontSize: '14px',
+    lineHeight: '20px',
+    letterSpacing: '0.25px',
+  },
+  emailChangeDescription: {
+    marginTop: '20px',
+    fontFamily: "'PT Sans', sans-serif",
+    fontSize: '12px',
+    lineHeight: '20px',
+    letterSpacing: '0.25px',
+  },
+  loginDescription: {
+    marginTop: '40px',
+    fontFamily: "'PT Sans', sans-serif",
+    fontSize: '16px',
+    lineHeight: '24px',
+    letterSpacing: '0.15px',
+  },
+  spacingMargin: {
+    marginTop: '40px',
+  },
+  logo: {
+    width: '114px',
+    height: '64px',
+  },
+  footerLogo: {
+    marginTop: '100px',
+    width: '215px',
+    height: '80px',
+  },
+  copyrightMsg: {
+    marginTop: '20px',
+    fontFamily: "'PT Sans', sans-serif",
+    fontSize: '10px',
+    lineHeight: '56px',
+  },
+})
+
+const Login = ({
+  classes,
+  product,
+  showPasscode,
+  emailChangeToggle,
+  passcodeResendHandler,
+  logo,
+  welcomeTitle,
+  welcomeDescription,
+  copyrightMessage,
+  loadingConfig,
+  errorConfig,
+  onEmailSubmit,
+  onPasscodeSubmit,
+  onChange,
+}) => {
+  const [email, setEmail] = useState('')
+  const [passcode, setPasscode] = useState('')
+  const [errors, setErrors] = useState({})
+  const [loading, setLoading] = useState({})
+
+  useEffect(() => {
+    if (loadingConfig) {
+      setLoading(loadingConfig)
+    }
+    if (errorConfig) {
+      setErrors(errorConfig)
+    }
+  }, [loadingConfig, errorConfig])
+
+  const handleEmailSubmit = () => {
+    if (!email) {
+      return setErrors((prev) => ({ ...prev, emailError: 'Email required' }))
+    }
+    if (onEmailSubmit) {
+      onEmailSubmit()
+    }
+  }
+
+  const handlePasscodeSubmit = () => {
+    if (!passcode) {
+      return setErrors((prev) => ({ ...prev, passcodeError: 'Passcode required' }))
+    }
+    if (onPasscodeSubmit) {
+      onPasscodeSubmit()
+    }
+  }
+
+  const handleChange = (field, value) => {
+    let changes = { email, passcode, errorConfig: errors, loadingConfig: loading }
+  
+    if (field === 'email') {
+      setEmail(value)
+      if (value && errors.emailError) {
+        setErrors((prev) => ({ ...prev, emailError: '' }))
+      }
+      if (!value) {
+        setErrors((prev) => ({ ...prev, emailError: 'Email required' }))
+        changes = { ...changes, errorConfig: { ...errors, emailError: 'Email required' } }
+      }
+    }
+  
+    if (field === 'passcode') {
+      setPasscode(value)
+      const match = (value).match(/[^A-F\d]/gi)
+
+      if (value && errors.passcodeError) {
+        setErrors((prev) => ({ ...prev, passcodeError: '' }))
+      }
+      if (!value || match) {
+        setErrors((prev) => ({ ...prev, passcodeError: match ? `'${[...match]}' is not a valid character for passcode` : 'Passcode required' }))
+        changes = { ...changes, errorConfig: { ...errors, passcodeError: match ? `'${[...match]}' is not a valid character for passcode` : 'Passcode required' } }
```

Comment:
Just a remarque... had to hide the new file tree in github to see this line. Can we perhaps agree to limit the line length to some more reasonable reading format? This one stands at 154 characters. The somewhat standard length would be 100-120.

Reply:
sounds good!